src/peer_manager.rs

```diff
@@ -212,15 +212,17 @@ pub struct Peer {
     state: PeerState,
     timestamp: Instant,
     valid: bool,
+    reconnect: bool,
```

Comment:
The peer manager is getting really confusing. Let's add a comment here to explain why this field exists.